src/fields2cover/route_planning/route_planner_base.cpp

```diff
@@ -125,52 +159,52 @@ F2CGraph2D RoutePlannerBase::createCoverageGraph(
     }
   }
 
-  F2CPoint deposit(-1e8, -1e8);  // Arbitrary point
-  if (this->r_start_end) {
-    deposit = *this->r_start_end;
+  F2CPoint deposit_start(-1e8, -1e8);  // Arbitrary point
+  if (this->r_start) {
+    deposit_start = *this->r_start;
   }
 
-  for (auto&& swaths : swaths_by_cells) {
-    for (auto&& s : swaths) {
-      if (this->r_start_end) {
-        g.addEdge(s.startPoint(), deposit, shortest_graph);
-        g.addEdge(s.endPoint(), deposit, shortest_graph);
-      } else {
-        g.addEdge(s.startPoint(), deposit, 0);
-        g.addEdge(s.endPoint(), deposit, 0);
-      }
-    }
+  g = addStartEndToGraph(cells, swaths_by_cells, shortest_graph, g,
+      deposit_start, this->r_start != std::nullopt);
+
+  F2CPoint deposit_end(-1e8, -1e8);  // Arbitrary point
+  if (this->r_end) {
+    deposit_end = *this->r_end;
   }
+  g = addStartEndToGraph(cells, swaths_by_cells, shortest_graph, g, deposit_end,
+      this->r_end != std::nullopt);
+
   return g;
 }
 
 std::vector<int64_t> RoutePlannerBase::computeBestRoute(
     const F2CGraph2D& cov_graph, bool show_log) const {
-  int depot_id = static_cast<int>(cov_graph.numNodes()-1);
+  int depot_id = static_cast<int>(cov_graph.numNodes() - 1);
```

Comment:
As you have added start and end points, now `depot_id` (the starting point) is not always located on the last point. If start and end are set, then `depot_id = cov_graph.numNodes()-2`

Reply:
Can you add tests on the route planner that show what happens if you set only the start, only the end, and what happens when start and end are extreme points of some swaths?

Reply:
I've added the code, but am unsure about the test you want added here.

Reply:
I've added a new branch called pr-177 with the test I was referring to. It crashes.

Reply:
Once that is solved, we can merge it.